tracer/src/Datadog.Trace/DataStreamsMonitoring/Aggregation/DataStreamsAggregator.cs

```diff
@@ -77,17 +113,40 @@ public void AddBacklog(in BacklogPoint point)
     /// <returns>True if data was serialized to the stream</returns>
     public bool Serialize(Stream stream, long maxBucketFlushTimeNs)
     {
-        var statsToAdd = Export(maxBucketFlushTimeNs) ?? new();
-        var backlogsToAdd = ExportBacklogs(maxBucketFlushTimeNs) ?? new();
-        if (statsToAdd.Count > 0 || backlogsToAdd.Count > 0)
+        Interlocked.Exchange(ref _pointsDropped, 0);
+        try
         {
-            _formatter.Serialize(stream, _bucketDurationInNs, statsToAdd, backlogsToAdd);
-            Clear(statsToAdd, backlogsToAdd);
+            _lock.AcquireWriterLock(_lockTimeoutMs);
+            try
+            {
+                var statsToAdd = Export(maxBucketFlushTimeNs) ?? new();
+                var backlogsToAdd = ExportBacklogs(maxBucketFlushTimeNs) ?? new();
+                if (statsToAdd.Count > 0 || backlogsToAdd.Count > 0)
+                {
+                    _formatter.Serialize(stream, _bucketDurationInNs, statsToAdd, backlogsToAdd);
+                    Clear(statsToAdd, backlogsToAdd);
 
-            return true;
-        }
+                    return true;
+                }
 
-        return false;
+                return false;
+            }
+            finally
+            {
+                var dropped = Interlocked.Read(ref _pointsDropped);
+                _lock.ReleaseWriterLock();
+
+                if (dropped > 0)
+                {
+                    Log.Debug("Dropped {Dropped} data points during serialization", dropped);
```

Comment:
This (and line 147 below) is using `Datadog.Trace.Vendors.Serilog.Log`, which is probably not what you want. Instead, the usual logging pattern is adding a static `IDatadogLogger` field to the type and using that field.
```csharp
private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<DataStreamsAggregator>();
```

You can find example usages by searching for `DatadogLogging.GetLoggerFor`.